packages/typespec-ts/test/integration/generated/internal/src/clientDefinitions.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetInternalParameters, PostInternalParameters } from "./parameters";
+import { GetInternal200Response, PostInternal200Response } from "./responses";
+import { Client, StreamableMethod } from "@azure-rest/core-client";
+
+export interface GetInternal {
+  get(options: GetInternalParameters): StreamableMethod<GetInternal200Response>;
+}
+
+export interface PostInternal {
+  post(
+    options: PostInternalParameters
+  ): StreamableMethod<PostInternal200Response>;
+}
+
+export interface Routes {
+  /** Resource for '/internal/getInternal' has methods for the following verbs: get */
+  (path: "/internal/getInternal"): GetInternal;
```

Comment:
I would suggest to differentiate the interal code and publish ones by document or naming convention.

Or we could enable a flag to not generate internal codes.

Reply:
For RLC we shouldn't do anything for internal. If it is in the Service Typespec it should be generated. AFAIK this is a feature that would be used in client.tsp to shape clients rather than for describing services. So for RLC we should just ignore this decorator.